docker/build/devdeps.Dockerfile

```diff
@@ -156,7 +156,7 @@ RUN apt-get update && apt-get install -y --no-install-recommends python3 python3
     && python3 -m pip install --no-cache-dir \
         ipython==8.15.0 pandoc==2.3 sphinx==5.3.0 sphinx_rtd_theme==1.2.0 sphinx-reredirects==0.1.2 \
         sphinx-copybutton==0.5.2 sphinx_inline_tabs==2023.4.21 enum-tools[sphinx] breathe==4.34.0 nbsphinx==0.9.2 sphinx_gallery==0.13.0 \
-     myst-parser==1.0.0
+     myst-parser==1.0.0 ipykernel==6.27.1 torch==2.1.1 torchvision==0.16.1
```

Comment:
I may be out of the loop on this one, but I wanted to make sure that @bettinaheim was aware of this upcoming dependency, especially since the `torch` wheel is  >600MB by itself, and when `pip install`-ed, its exploded wheel and exploded dependencies make the installed base grow by more than 4GB.
```bash
$ virtualenv /tmp/venv
$ du -hs /tmp/venv
11M     /tmp/venv
$ /tmp/venv/bin/pip install torch==2.1.1
...
$ du -hs /tmp/venv
4.6G    /tmp/venv
```

Reply:
I'm a little out of the loop on what extra validation steps have been added recently for python notebooks. But the docs for this PR have issues building due to some missing dependencies within the notebook itself. torch and torchvision are used in `examples/python/tutorials/executing_circuits.ipynb`, giving us the following error during docs build

```
sphinx.errors.SphinxParallelError: nbsphinx.NotebookError: CellExecutionError in examples/python/tutorials/hybrid_qnns.ipynb:

...

ModuleNotFoundError: No module named 'torchvision'

You can ignore this error by setting the following in conf.py:

    nbsphinx_allow_errors = True
```

An alternative to installing these packages would be to allow the errors, I imagine at the downside of ignoring build errors in every notebook, however. This issue will likely come up again in the future -- using packages in our notebook tutorials that we don't necessarily want/need as image dependencies -- so we should figure out a solid solution for this now